File: backend-csharp/Models/DanhGiaSanPham.cs
using System;
using System.Collections.Generic;

namespace WebFashion.Api.Models;

public partial class DanhGiaSanPham
{
    public long Id { get; set; }

    public long SanPhamId { get; set; }

    public long TaiKhoanId { get; set; }

    public long? DonHangId { get; set; }

    public int? SoSao { get; set; }

    public string? NoiDung { get; set; }

    public string? HinhAnh { get; set; }

    public string TrangThai { get; set; } = null!;

    public DateTime CreatedAt { get; set; }

    public long? ParentId { get; set; }

    public virtual DonHang? DonHang { get; set; }

    public virtual ICollection<DanhGiaSanPham> InverseParent { get; set; } = new List<DanhGiaSanPham>();

    public virtual DanhGiaSanPham? Parent { get; set; }

    public virtual SanPham SanPham { get; set; } = null!;

    public virtual TaiKhoan TaiKhoan { get; set; } = null!;

    public virtual ICollection<ThichDanhGium> ThichDanhGia { get; set; } = new List<ThichDanhGium>();
}

File: backend/models/DanhGiaSanPham.js
const { DataTypes } = require("sequelize");
const sequelize = require("../config/db");

const DanhGiaSanPham = sequelize.define(
  "DanhGiaSanPham",
  {
    id: { type: DataTypes.BIGINT, primaryKey: true, autoIncrement: true },
    san_pham_id: { type: DataTypes.BIGINT, allowNull: false },
    tai_khoan_id: { type: DataTypes.BIGINT, allowNull: false },
    don_hang_id: { type: DataTypes.BIGINT, allowNull: true },
    so_sao: { type: DataTypes.INTEGER, allowNull: true },
    noi_dung: { type: DataTypes.TEXT, allowNull: false },
    hinh_anh: { type: DataTypes.STRING(255), allowNull: true },
    trang_thai: { type: DataTypes.STRING(20), defaultValue: "pending" },
    parent_id: { type: DataTypes.BIGINT, allowNull: true },
  },
  {
    tableName: "DanhGiaSanPham",
    timestamps: true,
    createdAt: "created_at",
    updatedAt: false, // Bảng của bạn chỉ có created_at
  },
);

// Khai báo quan hệ với Tài Khoản (để lấy tên người đánh giá)
const TaiKhoan = require("./TaiKhoan");
DanhGiaSanPham.belongsTo(TaiKhoan, {
  foreignKey: "tai_khoan_id",
  as: "nguoi_dung",
});

module.exports = DanhGiaSanPham;
